Readme.md

```diff
@@ -91,6 +91,10 @@ A number of users have reported receiving the recommended diode board from vario
 
 The large triangular part of the diode, should be soldered to the positive side of the board not the negative. If yours is orientated as above, you should desolder the photodiode invert it and resolder so the larger triangular part of the diode is connected to positive.
 
+### Reduce the amount of data the sensors produce
+
+Depending on the `pulse_rate` variable, the type of house/apartment and the heating system in use, the sensors that are exposed to Home Assistant may produce a lot of data. For example, with the default `pulse_rate` in the file, 1000, a power consumption of 3600 W means that the sensors produce 2 HA state changes per second (which means 7200 state changes per hour). If you don't need that kind of granularity, you can use [ESPHome sensor filters](https://esphome.io/components/sensor/index.html#sensor-filters) filters to reduce the rate of updates written to Home Assistant. With the commented-out filters in [home_assistant_glow.yaml][file] enabled, only 396 state changes will be produced per hour.
```

Comment:
```suggestion
Depending on the `pulse_rate` variable, the type of house/apartment and the heating system in use, the sensors that are exposed to Home Assistant may produce a lot of data. For example, with the default `pulse_rate` of 1000 and a power consumption of 3600 W, means that the sensors produce 2 HA state changes per second (which means 7200 state changes per hour).

If you don't need that kind of granularity, you can use [ESPHome sensor filters](https://esphome.io/components/sensor/index.html#sensor-filters) filters to reduce the rate of updates written to Home Assistant. With the commented-out filters in [home_assistant_glow.yaml][file] enabled, only 396 state changes will be produced per hour.
```

Reply:
Maybe it should be a little more extensive with the described options that are now in yaml file and what consequences it has if you start using them? (although this is also described on the esphome website) 🤔

Reply:
True, that could make sense.

### W sensor

Regarding `throttle_average`, this is one consequence worth keeping in mind:

> no value(s) received: `NaN` is returned - add the heartbeat filter if periodical pushes are required and/or `filter_out: nan` if required

In my case and with `throttle_average: 10s`, anything less than 360 W power consumption would not be reported at all. I have a constant base load of about 400 W, so because of that, this hasn't been a problem for me 😁

Hm, are there any other filters or combinations of filters that could be used instead of `throttle_average`? Something that would reduce the amount of data, report somewhat accurate data, but still be able to report data for low amounts of power usage.

### kWh sensor

[This](https://github.com/klaasnicolaas/home-assistant-glow/pull/95#discussion_r772628731) is the only consequence I can think of:

> Restarting the ESPHome device may mean that up to < 0.1 kWh of data is lost too.

Reply:
I realized that using `on_value` for lighting up the ESP LED doesn't work together with the `throttle_average` filter. With `throttle_average`, `on_value` will just trigger every 10th second, and light up the LED on the ESP device even if the power meter LED hasn't blinked, and it will not blink more often than every 10th second either.

I think replacing it with `on_raw_value` should work: https://esphome.io/components/sensor/index.html#on-raw-value. And I think that `on_raw_value` should work fine even when `throttle_average` isn't used, right? So the default could be changed in that case.

Reply:
I haven't spent so much time on this lately, but I think this should be solved before we can consider this ready for merging. I'll try to get some time spent on in the near future.

Reply:
> In my case and with throttle_average: 10s, anything less than 360 W power consumption would not be reported at all. I have a constant base load of about 400 W, so because of that, this hasn't been a problem for me 😁
>
> Hm, are there any other filters or combinations of filters that could be used instead of throttle_average? Something that would reduce the amount of data, report somewhat accurate data, but still be able to report data for low amounts of power usage.

In the end this was embarrassingly easy 🙈 Just add a `filter_out: NaN` filter after the `throttle_average` filter.

Reply:
What do you think @klaasnicolaas, would it make sense to make these filters default? 🤔 I'm inclined to say it would. And maybe in that case make the delay configurable with a substitution.

Reply:
The text may have to change because from version 4.x you no longer put the pulse_rate in the yaml file, but in the UI.

Reply:
Good point, updated.

```diff
diff --git Readme.md Readme.md
index b936dd8..31bc7c7 100644
--- Readme.md
+++ Readme.md
@@ -119,7 +119,7 @@ The large triangular part of the diode, should be soldered to the positive side

 ### Reduce the amount of data the sensors produce

-Depending on the `pulse_rate` variable, the type of house/apartment and the heating system in use, the sensors that are exposed to Home Assistant may produce a lot of data. For example, with the default `pulse_rate` in the file, 1000, a power consumption of 3600 W means that the sensors produce 2 HA state changes per second (which means 7200 state changes per hour). If you don't need that kind of granularity, you can use [ESPHome sensor filters](https://esphome.io/components/sensor/index.html#sensor-filters) filters to reduce the rate of updates written to Home Assistant. With the commented-out filters in [home_assistant_glow.yaml][file] enabled, only 396 state changes will be produced per hour.
+Depending on the configured `pulse rate`, the type of house/apartment and the heating system in use, the sensors that are exposed to Home Assistant may produce a lot of data. For example, with the default `pulse rate` 1000, a power consumption of 3600 W means that the sensors produce 2 HA state changes per second (which means 7200 state changes per hour). If you don't need that kind of granularity, you can use [ESPHome sensor filters](https://esphome.io/components/sensor/index.html#sensor-filters) to reduce the rate of updates written to Home Assistant. With the commented-out filters in the [home_assistant_glow.yaml][file] enabled, only 396 state changes will be produced per hour.

 ### My Daily Energy won't reset
```